pkg/operator/controllers/muo/muo_controller.go

```diff
@@ -0,0 +1,128 @@
+package muo
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+)
+
+const (
+	enabled  string = "rh.srep.muo.enabled"
+	managed  string = "rh.srep.muo.managed"
+	pullSpec string = "rh.srep.muo.deploy.pullspec"
+)
+
+type Reconciler struct {
+	arocli        aroclient.Interface
+	kubernetescli kubernetes.Interface
+	dh            dynamichelper.Interface
+	deployer      *deployer
+
+	readinessPollTime time.Duration
+	readinessTimeout  time.Duration
+}
+
+func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, kubernetescli kubernetes.Interface, dh dynamichelper.Interface) *Reconciler {
+	return &Reconciler{
+		arocli:        arocli,
+		kubernetescli: kubernetescli,
+		dh:            dh,
+		deployer:      newDeployer(kubernetescli, dh),
+
+		readinessPollTime: 10 * time.Second,
+		readinessTimeout:  5 * time.Minute,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !instance.Spec.OperatorFlags.GetSimpleBoolean(enabled) {
+		// controller is disabled
+		return reconcile.Result{}, nil
+	}
+
+	// the trusted-ca-bundle being injected shouldn't cause a reconcile
+	if request.Name == "trusted-ca-bundle" && request.Namespace == "openshift-managed-upgrade-operator" {
+		return reconcile.Result{}, nil
+	}
```

Comment:
It's to prevent unneeded reconcilation.

Since OpenShift is responsible for maintaining this (by injecting the trust bundle), we don't want to respond to its reconciliation by OpenShift, I think.